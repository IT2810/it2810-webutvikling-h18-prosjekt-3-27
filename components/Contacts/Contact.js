import React, {Component} from "react";
import {FlatList, StyleSheet, Text, TouchableOpacity, ScrollView, View,} from "react-native";
import Icon from "react-native-vector-icons/Ionicons";


export default class Contact extends Component {
  constructor(props) {
    super(props);
  }

  render () {
    return (
      <View style={styles.contactList}>
        <View>
          <Icon name="ios-person" color={this.props.contact.color} size={50}/>
        </View>
        <View style={styles.contact}>
          <Text style={{color: this.props.contact.color}}>
            {this.props.contact.name}
          </Text>
          <Text style={styles.number}>
            {this.props.contact.number}
          </Text>
        </View>
      </View>
    )
  }
}

const styles = StyleSheet.create({
  contactList: {
    margin: 10,
    flexDirection: 'row',
  },

  contact: {
    paddingTop: 7,
    paddingLeft: 5,
  },
  number: {
    color: 'grey',
  },
});