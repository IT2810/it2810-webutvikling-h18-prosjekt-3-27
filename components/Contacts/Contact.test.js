import React from 'react';
import Contact from './Contact';
import ShallowRenderer from "react-test-renderer/shallow";



test('renders correctly', () => {
  const testContact = {
    name: 'mama',
    number: '1234',
    key: '1',
    color: 'green'
  };

  const renderer = new ShallowRenderer();
  const tree = renderer.render(<Contact
    contact = {testContact}
  />);
  expect(tree).toMatchSnapshot();
});