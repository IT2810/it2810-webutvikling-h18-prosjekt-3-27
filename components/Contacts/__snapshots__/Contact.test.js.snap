// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  style={
    Object {
      "flexDirection": "row",
      "margin": 10,
    }
  }
>
  <Component>
    <Icon
      allowFontScaling={false}
      color="green"
      name="ios-person"
      size={50}
    />
  </Component>
  <Component
    style={
      Object {
        "paddingLeft": 5,
        "paddingTop": 7,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "green",
        }
      }
    >
      mama
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "grey",
        }
      }
    >
      1234
    </Text>
  </Component>
</Component>
`;
